### ISSUE TITLE
Callback function Errors with homebridge-node-alarm-dot-com plugin

### ISSUE THREAD

**Issue by wesbro** [hoobs-org/HOOBS#442]
I've been working with the HOOBS team and a plugin developer to isolate an issue related to a recent plugin upgrade. At the beginning of the issue I was running Homebridge version 0.4.50 (on a HOOBS box) and node version 10.17.0. I was running the plugin homebridge-node-alarm-dot-com release 1.3.2. When I upgraded to the latest release (1.4.1) everything started to break down.

When I upgraded to homebridge-node-alarm-dot-com release 1.4.1 I began receiving messages that the "Callback function has already been called by someone else." However, this was only occurring on lights and not on locks. The plugin developer suggested I upgrade the node. HOOBS support suggested I run the following command: 'wget -q -O - http://bit.ly/get-hoobs | sudo bash /dev/stdin --node 12.14.0'

After I tried to update the node, I received this message: https://a.cl.ly/L1ukvQr1. The box was basically unusable at that point so HOOBS support reset my box to factory settings except for Node, which they upgraded to 12.14.1. I reinstalled the two plugins I was running before homebridge-node-alarm-dot-com (release 1.4.1) and Nest. Everything was running the same as it did when I did the initial plugin upgrade -- still received "Callback function has already been called by someone else" in the logs (again, only with the lights -- locks work fine). I should also note that the plugin tends to state the correct state of the lights (ie. if its on or off), but only has errors when the command is run. Here's a screenshot of those logs: https://a.cl.ly/WnuEp5d1

I attempted to rollback to the 3 most recent versions of the plugin using 'npm install homebridge-node-alarm-dot-com@1.3.2' and replacing the version number each time. It didn't seem to matter which version I was on -- results were the same. I'm  wondering if something was left over between when it broke a few days ago, and when it was repaired recently. 

I searched some forums and there was a suggestion that I could have another instance of Homebridge running, It said I should run 'ps aux | grep homebridge' to make sure I didn't already have an instance of homebridge running in the background. It came back with this, which I don't know if that means I'm running more than one version: https://a.cl.ly/12uDQQkm

**Comment by mkellsy** [comment]
The process is this.
1. Make sure the plugin is installed on the plugins screen. If not install it there. That properly links the plugin.

2. From the terminal run `cd ~/.hoobs`

2. Then run `npm install homebridge-node-alarm-dot-com@1.3.2`

If you wish to run separate instances, it is tougher but it is explained here

https://github.com/hoobs-org/HOOBS/wiki/6.2-Command-Line-Interface#running-a-cluster-client

And here

https://github.com/hoobs-org/HOOBS/wiki/6.2-Command-Line-Interface#running-a-cluster-instance

**Comment by wesbro** [comment]
@mkellsy Thanks - I just ran these to double check that I downgraded properly. And I am back at 1.3.2. However, I wasn't suggesting that I _wanted_ to run separate instances. The forum comment I found on the matter suggested that running more than one instance of homebridge could be the cause for the callback function error. I wasn't sure if that was something I might have done accidentally -- because I would have no way of knowing if that was something I stumbled into.
